src/Data/Array/Accelerate/AST.hs

```diff
@@ -748,6 +748,10 @@ data PrimFun sig where
   PrimLOr  :: PrimFun ((PrimBool, PrimBool) -> PrimBool)
   PrimLNot :: PrimFun (PrimBool             -> PrimBool)
 
+  -- local array operators
```

Comment:
These ops shouldn't be in `PrimFun` actually, they should just be part of `Exp` and replace the more specialised and now redundant `VecPack` and `VecUnpack`. (We should double check that the generated llvm-ir is the same before deleting them but iirc it generates a sequence of `InsertElement` / `ExtractElement` instructions)

Updating the pretty printer to detect the current pack/unpack case (where all the indices are constant) and displaying that in the current way would be good too.